components/channel_header/channel_header.js

```diff
@@ -124,6 +125,10 @@ class ChannelHeader extends React.PureComponent {
         if (header !== prevHeader) {
             this.props.actions.getCustomEmojisInText(header);
         }
+
+        if (this.props.channel.id !== prevProps.channel.id) {
+            this.props.actions.setRhsExpanded(false);
```

Comment:
This should be done in the SidebarRight component since that's the one that cares about whether or not the sidebar is expanded or not. It's likely to get lost if it's left here since others wouldn't expect to find sidebar logic here.

Reply:
Cool. I'll have a look at it.